samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -65,13 +65,7 @@
   private volatile String ephemeralPath = null;
   private final ZkKeyBuilder keyBuilder;
   private final int connectionTimeoutMs;
-  private ZkJobCoordinatorMetrics metrics;
-
-  public ZkUtils(ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs) {
-    this.keyBuilder = zkKeyBuilder;
-    this.connectionTimeoutMs = connectionTimeoutMs;
-    this.zkClient = zkClient;
-  }
+  private final ZkJobCoordinatorMetrics metrics;
 
   public ZkUtils(ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs, ZkJobCoordinatorMetrics metrics) {
```

Comment:
It will be cleaner to simply pass the metrics registry associated with this component and register more granular group of metrics under `ZkUtilsMetrics`. Overloading it with `ZkJobCoordinatorMetrics` is confusing as this component - ZkUtils is also accessed from CoordinationService. 

If you pass MetricsRegistry into this constructor, it can work for CoordinationService as well, whenever we get to completing SAMZA-1153 (pull out entire metric reporter lifecycle into localapplicationrunner). For now, it seems like we should be creating ZkUtils metrics as a subset of ZkJobCoordinatorMetrics and register it with Metricsregistry provided.

```
public ZkUtils (ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs, MetricsRegistry metricsRegistry) {
...
this.zkUtilsMetrics = new ZkUtilsMetrics();
metricsRegistry.register(this.zkUtilsMetrics);
...
}
```

If you have more questions on why we are passing around MetricsRegistry and the pattern used to de-couple a component's metrics instance from the reporting itself, take a look at the design in [dropwizard.io](https://github.com/dropwizard/dropwizard)
HTH. Thanks!